src/components/sync-connected-sites.tsx

```diff
@@ -17,43 +17,81 @@ interface ConnectedSiteSection {
 }
 
 export function SyncConnectedSites( {
-	syncSites,
 	connectedSites,
 	openSitesSyncSelector,
 	disconnectSite,
 }: {
-	syncSites: SyncSite[];
 	connectedSites: SyncSite[];
 	openSitesSyncSelector: () => void;
 	disconnectSite: ( id: number ) => void;
 } ) {
 	const { __ } = useI18n();
 	const siteSections: ConnectedSiteSection[] = useMemo( () => {
 		const siteSections: ConnectedSiteSection[] = [];
+		const processedSites = new Set< number >();
 
 		connectedSites.forEach( ( connectedSite ) => {
-			const syncSite = syncSites.find( ( site ) => site.id === connectedSite.id );
-			if ( syncSite ) {
-				const section: ConnectedSiteSection = {
-					id: syncSite.id,
-					name: syncSite.name,
-					provider: 'wpcom',
-					connectedSites: [ connectedSite ],
-				};
-				if ( syncSite.stagingSiteIds ) {
-					for ( const id of syncSite.stagingSiteIds ) {
-						const stagingSite = syncSites.find( ( site ) => site.id === id );
-						if ( stagingSite ) {
-							section.connectedSites.push( stagingSite );
-						}
+			if ( processedSites.has( connectedSite.id ) ) {
+				return; // Skip if we've already processed this site
+			}
+
+			const section: ConnectedSiteSection = {
+				id: connectedSite.id,
+				name: connectedSite.name,
+				provider: 'wpcom',
+				connectedSites: [ connectedSite ],
+			};
+
+			processedSites.add( connectedSite.id );
+
+			if ( connectedSite.stagingSiteIds ) {
+				for ( const id of connectedSite.stagingSiteIds ) {
+					const stagingSite = connectedSites.find( ( site ) => site.id === id );
+					if ( stagingSite ) {
+						section.connectedSites.push( stagingSite );
+						processedSites.add( stagingSite.id );
 					}
 				}
-				siteSections.push( section );
 			}
+
+			siteSections.push( section );
 		} );
 
 		return siteSections;
-	}, [ connectedSites, syncSites ] );
+	}, [ connectedSites ] );
+
+	const handleDisconnectSite = async ( sectionId: number, sectionName?: string ) => {
+		const dontShowDisconnectWarning = localStorage.getItem( 'dontShowDisconnectWarning' );
+		if ( ! dontShowDisconnectWarning ) {
+			const CANCEL_BUTTON_INDEX = 1;
+			const DISCONNECT_BUTTON_INDEX = 0;
+
+			const disconnectMessage = sectionName
+				? __( 'Disconnect' ) + ' ' + sectionName
```

Comment:
Let's use `sprintf` instead of concat two strings. That will improve the experience for RTL languages.

```suggestion
				? sprintf( __( 'Disconnect %s' ), sectionName )
```

Reply:
sounds good, I applied the suggestion https://github.com/Automattic/studio/pull/610/commits/90de8b01bcb0ae6e8dc245638c35741cf15f37b0